netty/src/main/java/io/grpc/transport/netty/NettyServerHandler.java

```diff
@@ -287,6 +289,12 @@ private void sendResponseHeaders(ChannelHandlerContext ctx, SendResponseHeadersC
     encoder().writeHeaders(ctx, cmd.streamId(), cmd.headers(), 0, cmd.endOfStream(), promise);
   }
 
+  private void cancelStream(ChannelHandlerContext ctx, CancelServerStreamCommand cmd,
+      ChannelPromise promise) {
+    serverStream(requireHttp2Stream(cmd.streamId())).abortStream(cmd.reason(), false);
```

Comment:
`notifyClient = false` is fine here. Since we are informing the client with the `writeRstStream()`, there is not a need for `abortStream()` to do so. As seen in the documentation for `notifyClient`: "if the stream is still writable" (and we could reverse the order of the calls, if we wanted to be strict about it :) ).

However, using `false` _should_ be an optimization; at least I think that's why the option was added. Using `false` was a "don't even bother." If it fails (instead of just doing useless work), then please point it out, because it is likely a bug.

The comment is definitely a "investigate" comment. I expect we need the functionality provided with the argument in some form in some cases. We should probably still investigate it by auditing the usages today. But it is also working reasonably already.